firefox-ios/firefox-ios-tests/Tests/ClientTests/TermsOfUse/TermsOfUseTelemetryTests.swift

```diff
@@ -8,126 +8,162 @@ import Glean
+    // Mock wrapper to capture telemetry calls without depending on Glean framework
+    private var gleanWrapper: MockGleanWrapper!
 
     override func setUp() {
         super.setUp()
-        telemetry = TermsOfUseTelemetry()
-        setupTelemetry(with: MockProfile())
+
+        // Injecting mock telemtry wrapper instead of real Glean
+        gleanWrapper = MockGleanWrapper()
+        telemetry = TermsOfUseTelemetry(gleanWrapper: gleanWrapper)
     }
 
     override func tearDown() {
         super.tearDown()
         telemetry = nil
-        tearDownTelemetry()
+        gleanWrapper = nil
     }
 
+    // Verifies: Display event + impression counter increment
     func testTermsOfUseBottomSheetDisplayed() throws {
         telemetry.termsOfUseDisplayed(surface: .bottomSheet)
-        let events = try XCTUnwrap(GleanMetrics.TermsOfUse.shown.testGetValue())
-        XCTAssertEqual(events.count, 1)
-        let event = events[0]
-        XCTAssertEqual(event.extra?["surface"], TermsOfUseTelemetry.Surface.bottomSheet.rawValue)
-        XCTAssertEqual(event.extra?["tou_version"], String(telemetry.termsOfUseVersion))
-
-        // Test impression counter
-        let impressionCount = try XCTUnwrap(GleanMetrics.UserTermsOfUse.shownCount.testGetValue())
-        XCTAssertEqual(impressionCount, 1)
+
+        let event = GleanMetrics.TermsOfUse.shown
+        let savedEvent = try XCTUnwrap(
+            gleanWrapper.savedEvents.first as? EventMetricType<GleanMetrics.TermsOfUse.ShownExtra>)
+        let savedExtras = try XCTUnwrap(
+            gleanWrapper.savedExtras.first as? GleanMetrics.TermsOfUse.ShownExtra)
+
+        XCTAssert(savedEvent === event, "Received \(savedEvent) instead of \(event)")
+        XCTAssertEqual(gleanWrapper.recordEventCalled, 1)
+        XCTAssertEqual(savedExtras.surface, TermsOfUseTelemetry.Surface.bottomSheet.rawValue)
+        XCTAssertEqual(savedExtras.touVersion, String(telemetry.termsOfUseVersion))
+        XCTAssertEqual(gleanWrapper.incrementCounterCalled, 1)
     }
 
+    // Verifies: Acceptance event + recorded Tou version + acceptance timestamp
     func testTermsOfUseAcceptButtonTapped() throws {
-        telemetry.termsOfUseAcceptButtonTapped(surface: .bottomSheet, acceptedDate: Date())
-
-        // Test accepted event
-        let events = try XCTUnwrap(GleanMetrics.TermsOfUse.accepted.testGetValue())
-        XCTAssertEqual(events.count, 1)
-        let event = events[0]
-        XCTAssertEqual(event.extra?["surface"], TermsOfUseTelemetry.Surface.bottomSheet.rawValue)
-        XCTAssertEqual(event.extra?["tou_version"], String(telemetry.termsOfUseVersion))
-        // Test version metric
-        let version = try XCTUnwrap(GleanMetrics.UserTermsOfUse.versionAccepted.testGetValue())
-        XCTAssertEqual(version, telemetry.termsOfUseVersion)
-        // Test date metric
-        let dateMetric = try XCTUnwrap(GleanMetrics.UserTermsOfUse.dateAccepted.testGetValue())
```

Comment:
I notice these were missing in the new tests, curious why we decided to leave testing the metric values out.